crates/auth/src/api/handlers/auth.rs

```diff
@@ -295,18 +296,25 @@ pub async fn refresh_token_handler(
 
     // Check node URL if token has node information
     if let Some(token_node_url) = &refresh_claims.node_url {
```

Comment:
DRY: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

Let's make this a helper method instead.

Also we can use: https://doc.rust-lang.org/rust-by-example/error/option_unwrap/and_then.html

```
 if let Some(host_header) = headers.get("host")
        .and_then(|h| h.to_str().ok())
        .and_then(|host| {
            Url::parse(token_node_url).ok()
                .and_then(|token_url| token_url.host_str().map(|token_host| (host, token_host)))
        }) {
```

to simplify the code a bit.